tools/codegen/api/types.py

```diff
@@ -394,5 +400,21 @@ def from_schema(func: FunctionSchema) -> 'NativeSignature':
             _returns_type=returns_type,
         )
 
+# ------------------------------------------------------------------- #
+
+#                           meta api
+
+# ------------------------------------------------------------------- #
+
+@dataclass(frozen=True)
+class MetaArgument:
+    type: str
+    name: str
+    # By fiat, meta argument functions must be on full c10 dispach
```

Comment:
basically, there is always a one-to-one correspondence between JIT argument (Argument) and meta argument. This is true because I am going to explicitly not support structured kernels for function signatures who cannot support `use_c10_dispatcher: full`